src/kOS/Serialization/ConfigNodeFormatter.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using kOS.Safe.Serialization;
+using System.Collections.Generic;
+using kOS.Safe;
+using kOS.Safe.Utilities;
+using System.Linq;
+using System.Collections;
+using kOS.Safe.Encapsulation;
+
+namespace kOS.Serialization
+{
+    public class ConfigNodeFormatter : IFormatWriter, IFormatReader
+    {
```

Comment:
Does this class guarantee that none of the content inside the data dumped to the config node will break config node syntax?  I know there's a few characters that you can't have in a string value in a ConfigNode because they confuse the ConfigNode format itself.  We ran into this problem when trying to save file content in local volumes, which is why the file contents get base64 encoded when storing them in the ConfigNode.  I did a quick scan by eye and didn't see any similar protection here, but then again I might have missed it if it was buried inside the Dump methods or serialization methods themselves.

Specifically, I'm thinking about when you dump strings to the ConfigNode.  Since the strings are open to the user putting whatever they like into them via the `CHAR()` function, we can't make any assumptions about their content being safe.